src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserService.java

```diff
@@ -27,14 +27,17 @@ public EnvelopeFinaliserService(
     }
 
     @Transactional
-    public void finaliseEnvelope(UUID envelopeId) {
+    public void finaliseEnvelope(UUID envelopeId, Long ccdId, String envelopeCcdAction) {
```

Comment:
it could take `ProcessedEnvelope` instead of 3 params

Reply:
First I thought the same but testing is much more clean and easy when using params.